apps/pipelines/nodes/base.py

```diff
@@ -297,20 +326,30 @@ def _process(self, state: PipelineState) -> PipelineState:
     """
 
     def process(
-        self, incoming_edges: list, outgoing_edges: list, state: PipelineState, config: RunnableConfig
-    ) -> PipelineState:
+        self, incoming_nodes: list, outgoing_nodes: list, state: PipelineState, config: RunnableConfig
+    ) -> PipelineState | Command:
         self._config = config
-        state = self._prepare_state(self.node_id, incoming_edges, state)
+        self._incoming_nodes = incoming_nodes
+        self._outgoing_nodes = outgoing_nodes
+        state = PipelineState(state)
```

Comment:
Is `state` a dict here? The type hint says it's `PipelineState`, but I suppose that wrong then?